Source/Orts.Simulation/Common/Scripting/ScriptManager.cs

```diff
@@ -34,7 +34,7 @@ public class ScriptManager
     {
         readonly Simulator Simulator;
         readonly Dictionary<string, Assembly> Scripts = new Dictionary<string, Assembly>();
-        static readonly CSharpCodeProvider Compiler = new CSharpCodeProvider(new Dictionary<string, string>() { { "CompilerVersion", "v3.5" } });
+        static readonly CSharpCodeProvider Compiler = new CSharpCodeProvider(new Dictionary<string, string>() { { "CompilerVersion", "v4.0" } });
```

Comment:
Technically the provided change fixes the issue for now.
The ultimate fix would be to remove the version parameter at all, so the code compiles in the same version on default as the program is running on. Else we may run into the same issue again at some point when compiler version changes.
```suggestion
        static readonly CSharpCodeProvider Compiler = new CSharpCodeProvider();
```

Reply:
Hi Steffen,
yes, what you say is correct. Take into account anyhow that MS has already changed its policy, because if you write "4.0" it's valid also for all 4.x versions. In fact if you write "4.5" or "4.7.2" you get an error.